CMakeLists.txt

```diff
@@ -18,6 +18,10 @@ add_subdirectory(lib/bls-signatures)
 
 find_package(Threads REQUIRED)
 
+if(APPLE)
+find_package(sodium REQUIRED)
```

Comment:
maybe this makes sense in genera, not just apple?

Reply:
You can add it on other OS, but you have to manually add `FindSodium.cmake` to the CMake modules folder, on macOS it is in `/usr/local/opt/cmake/share/cmake/Modules/`.

On other OSs, if you add `find_package(sodium REQUIRED)`, you have to add `FindSodium.cmake` manually too. In fact, even if you don't add `find_package(sodium REQUIRED)`, it can be compiled successfully.

Only on macOS, it is necessary to add `find_package(sodium REQUIRED)`, so I add `if(APPLE)`.